UnityProject/Assets/Scripts/Items/Science/Implants/ReviverImplant.cs

```diff
@@ -0,0 +1,48 @@
+using UnityEngine;
+using HealthV2;
+using AddressableReferences;
+
+namespace Items.Implants.Organs
+{
+	public class ReviverImplant : BodyPartFunctionality
+	{
+
+		[SerializeField] private UniversalObjectPhysics objectPhysics;
+
+		[SerializeField] private AddressableAudioSource reviveSound;
+
+		[SerializeField] private int delaySeconds = 600;
+
+		private float lastTrigger = 0; 
+
+		public override void AddedToBody(LivingHealthMasterBase livingHealth)
+		{
+			lastTrigger = Time.time - delaySeconds; //Ready to revive
+			RelatedPart.HealthMaster.OnTakeDamageType += Revive;
+		}
+
+		public override void RemovedFromBody(LivingHealthMasterBase livingHealth)
+		{
+			RelatedPart.HealthMaster.OnTakeDamageType -= Revive;
+		}
+
+		public void Revive(DamageType damageType, GameObject source)
```

Comment:
might be better to put this on update loop instead? because this would only update when damage to the ReviverImplant happens

Reply:
Oh yeah, no it was supposed to be in update, I must of forgotten to change it